src/modules/m_restrictchans.cpp

```diff
@@ -27,13 +27,38 @@ typedef insp::flat_set<std::string, irc::insensitive_swo> AllowChans;
 class ModuleRestrictChans : public Module
 {
 	AllowChans allowchans;
+	UserModeReference regmode;
+	bool allowregistered;
+
+	bool CanCreateChannel(LocalUser* user, const std::string& name)
+	{
+		if (allowregistered && user->IsModeSet(regmode))
```

Comment:
Checking if the user has an account name is a more reliable way of checking that users are logged in.

Reply:
That's true, I copied that logic from `regonlycreate` but thinking about it, yeah that would only match if the user is actively using their registered nickname instead of just being logged in. Will fix.